### DISCUSSION TITLE
Creating A Circle with a 2 x 2 Center

### DISCUSSION THREAD

**Discussion by NoSignalBeep** [maruohon/minihud#315] [General]
Hello, I would like to make a circle around an Ocean Monument, but it has a 2 x 2 center. How would I go about doing that when it comes to setting the center so that it is symmetrical all the way around? What is the difference between block snap corner, center, and none? What does combine quads mean? I've tried watching a tutorial, but the user only used a center block and created an odd-numbered circle.

**Comment by maruohon** [comment]
The block snap settings are pretty much what they sound like. So if you imagine the center point being a single point, that tells you where within a single block that center will be.

So for an even sized diameter you want it in the corner, so that there is a full number of blocks on each side, instead of the half block there would be if it's in the center of a block. So basically imagine it in the center intersection of 4 blocks in the `Corner` setting.

The `None` setting allows you to freely set the center point anywhere, not automatically snapped to any block-relative position.

The `Combine Quads` option is just for rendering, basically combining "aligned" rectangles into bigger rectangles, which improves rendering performance. You can easily see what it does by flipping it on and off. It will build as big rectangles as it can, instead of rendering each block position individually. Whether you need or want that visual look is up to you. Usually you don't "need" it, even full despawn spheres shouldn't be *super horrible* for performance, although they definitely have a noticeable impact already.

**Reply by NoSignalBeep** [reply]
I ran a test. I laid out my 2x2 square and laid blocks out 50 blocks from the center line in each direction. I then created one circle from each corner of a distinctively different color and found one that was symmetrical. I couldn't for the life of me to get it to match up. The magic corner ended up being the south east corner. Not sure why, but I'm keen as a peach about it. I get tired of going back and forth on the circle website because I mess up sometimes or don't line it up right and this visual aid is really helpful. Also for knowing which terrain I still need to clear. I tore down the monument to prepare to drain with sweepers without paying attention to my surroundings.

**Reply by maruohon** [reply]
I'm sorry but I didn't quite get if you managed to line it up like you wanted, or was the circle crooked/non-symmetrical?

I'm not entirely sure about the circle algorithm, but at least the sphere shapes are currently a bit broken/crooked ever since the 1.18 update when I rewrote the code partially. I thought it would only affect the spheres, but I can't remember for sure. The spheres do use the circle code as a base to build "rings", but I *think* I checked it at one point and the circle was symmetrical... but I have a terrible memory for any things so I really can't remember anymore.

**Reply by NoSignalBeep** [reply]
I did manage to get it lined up, but it took me rendering 4 circles with 4 different colors to compare. I also added 50 blocks in each direction from the center line for the radius to be visual. I stood on each corner and rendered a new shape with a new color. Then I swam around and found a color that lined up on all four areas. It ended up being the south east corner. I don't know if that's significant information or not, but that's how I figured it out.

**Comment by NoSignalBeep** [comment]
Wow, thank you so much. I've never gotten an answer so quickly on GitHub before. I really appreciate it, and I really like this mod. I'm learning more about it all the time, and it has helped tremendously in farm rates and such. For instance, a lot of people get afk spots wrong in videos because I don't think they utilize the tools available to them, or don't know how to. You make it easier, and I want to thank you. If I had the funds, I would donate, but unfortunately life's circumstances provide me with a fixed income. That may be too much information, but I'm very transparent. You have a nice day.
